scripts/packages/VSCodeServer/src/request_handlers.jl

```diff
@@ -0,0 +1,143 @@
+                @eval Main ans = $(QuoteNode(ans))
+            catch err
+                EvalError(err, catch_backtrace())
+            end
+
+            if show_result
+                if res isa EvalError
+                    Base.display_error(stderr, res.err, res.bt)
+                elseif res !== nothing && !ends_with_semicolon(source_code)
+                    Base.invokelatest(display, res)
+                end
+            else
+                try
+                    Base.invokelatest(display, InlineDisplay(), res)
+                catch err
+                    if !(err isa MethodError)
+                        printstyled(stderr, "Display Error: ", color = Base.error_color(), bold = true)
+                        Base.display_error(stderr, err, catch_backtrace())
+                    end
+                end
+            end
+
+            rendered_result = safe_render(res)
+        end
+    end
+    return rendered_result
+end
+
+function repl_getvariables_request(conn, params::Nothing)
+    M = Main
+    variables = []
+    clear_lazy()
+
+    for n in names(M, all=true, imported=true)
+        !isdefined(M, n) && continue
+        Base.isdeprecated(M, n) && continue
+
+        x = getfield(M, n)
+        x === vscodedisplay && continue
+        x === VSCodeServer && continue
+        x === Main && continue
+
+        n_as_string = string(n)
+        startswith(n_as_string, "#") && continue
+        t = typeof(x)
+
+        rendered = treerender(x)
+
+        push!(variables, ReplGetVariablesRequestReturn(
+            string(t),
+            get(rendered, :head, "???"),
+            n_as_string,
+            get(rendered, :id, get(get(rendered, :child, Dict()), :id, false)),
+            get(rendered, :haschildren, false),
+            get(rendered, :lazy, false),
+            get(rendered, :icon, ""),
+            can_display(x)
+        ))
+    end
+
+    return variables
+end
+
+function repl_getlazy_request(conn, params::Int)
+    res = get_lazy(params)
+
+    return res
+end
+
+function repl_showingrid_notification(conn, params::String)
+    try
+        var = Core.eval(Main, Meta.parse(params))
+
+        Base.invokelatest(internal_vscodedisplay, var)
+    catch err
+        Base.display_error(err, catch_backtrace())
+    end
+end
+
+function repl_loadedModules_request(conn, params::Nothing)
+    res = string.(collect(get_modules()))
+
+    return res
+end
+
+function repl_isModuleLoaded_request(conn, params::String)
+    is_loaded = is_module_loaded(params)
+
+    return is_loaded
+end
+
+function repl_startdebugger_request(conn, params::String, crashreporting_pipename)
+    hideprompt() do
+        debug_pipename = params
+        try
+            DebugAdapter.startdebug(debug_pipename)
+        catch err
+            DebugAdapter.global_err_handler(err, catch_backtrace(), crashreporting_pipename, "Debugger")
+        end
+    end
+end
```

Comment:
How about moving each request handler so that they live close to its logic, rather than aggregating them here ?
E.g.
- `repl_loadedModules_request` into module.jl
- `repl_getvariables_request` into trees.jl (which actually I prefer to rename to workspace.jl or something if we go this way)

In other word we will organize files based on feature rather than internal design. Juno also organizes handlers in this way: e.g.
https://github.com/JunoLab/Atom.jl/blob/75eebb773ec222368801968ff5d6ae1e083c4fbf/src/workspace.jl#L13

Reply:
Yes, I think that would be even nicer. I'd say in a follow up PR?

Reply:
Yeah, maybe I will.